mod/chain-spec/pkg/chain/chain_spec.go

```diff
@@ -207,14 +218,29 @@ func NewChainSpec[
 	CometBFTConfigT any,
 ](data SpecData[
 	DomainTypeT, EpochT, ExecutionAddressT, SlotT, CometBFTConfigT,
-]) Spec[
+]) (Spec[
 	DomainTypeT, EpochT, ExecutionAddressT, SlotT, CometBFTConfigT,
-] {
-	return &chainSpec[
+], error) {
+	c := &chainSpec[
 		DomainTypeT, EpochT, ExecutionAddressT, SlotT, CometBFTConfigT,
 	]{
 		Data: data,
 	}
+	return c, c.validate()
+}
+
+// validate ensures that the chain spec is valid, returning error if it is not.
+func (c *chainSpec[
+	DomainTypeT, EpochT, ExecutionAddressT, SlotT, CometBFTConfigT,
+]) validate() error {
+	if c.MaxWithdrawalsPerPayload() <= 1 {
+		return ErrInsufficientMaxWithdrawalsPerPayload
+	}
+
+	// EVM Inflation values can be zero or non-zero, no validation needed.
+
+	// TODO: Add more validation rules here.
```

Comment:
yes, good we have the scaffolding to do this rn.
Note: these checks may be fork dependent too